python_ta/typecheck/base.py

```diff
@@ -8,6 +8,28 @@
 from ..util.monad import Failable, failable_collect
 
 
+class _TNode:
+    """A node in the TypeConstraints disjoint set data structure."""
+    type: type
+    parent: Optional['_TNode']
+    adj_list: List[Tuple['_TNode', NodeNG]]
+    ast_node: Optional[NodeNG]
+
+    def __init__(self, node_type: type, ast_node: Optional[NodeNG] = None):
+        self.type = node_type
+        self.parent = None
+        self.adj_list = []
+        self.ast_node = ast_node
+
+    def __eq__(self, other):
```

Comment:
Add a type signature here. You can assume `other` is a `_TNode`, so don't check it below.